add events

auryn-macmillan opened the pull request:
This PR adds events to the Reality contracts.

Closes #75

cristovaoth reviewed (approved):
Pretty solid and straightforward code changes, along with the introduction of events. 👌

manboy-eth reviewed (commented):
Nice, it will be much easier to view the interactions off-chain now.

Added some comments.

Also, is it possible to add an event for when a proposal is marked as invalid? So that we know we do not have to watch for that proposalId anymore and can mark it as invalid (even if it has not expired).

auryn-macmillan commented on contracts/RealityModule.sol:
```diff
@@ -24,13 +24,20 @@ abstract contract RealityModule is Module {
         string indexed proposalId
     );
-
+    event ProposalExecuted(string proposalId, bytes32 txHash, uint256 txIndex);
```
Done!

auryn-macmillan commented on contracts/RealityModule.sol:
```diff
@@ -24,13 +24,20 @@ abstract contract RealityModule is Module {
+    event SetAnswerExpiration(uint32 answerExpiration);
+    event SetArbitrator(address indexed questionArbitrator);
+    event SetMinimumBond(uint256 minimumBond);
+    event SetTemplate(uint256 templateId);
```
Done!

auryn-macmillan commented on contracts/RealityModule.sol:
```diff
@@ -24,13 +24,20 @@ abstract contract RealityModule is Module {
         bytes32 indexed questionId,
         string indexed proposalId
```
Sure!

auryn-macmillan commented:
Ok, I've added the extra events.

Given that I added extra functions, these changes are going to need to be audited. That being the case, I've also bumped the solidity version to 0.8.6 and added custom errors.

I'll hold off on merging this until it has been audited.
